File: backend/Models/Subscription.cs
namespace MechanicApp.Server.Models
{
    public class Subscription
    {
        public int Id { get; set; }
        public string Email { get; set; } = string.Empty;
        public string? StripeSessionId { get; set; }
        public string? StripeSubscriptionId { get; set; }

        /// <summary>active, inactive, cancelled, refunded, expired</summary>
        public string Status { get; set; } = "inactive";

        public string? PlanName { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? ExpiresAt { get; set; }
        public string? StripePayload { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: frontend/src/app/models/subscription.ts
export interface SubscriptionStatus {
  active: boolean;
  status: string;
  planName?: string;
  expiresAt?: string;
  email?: string;
  isDemo?: boolean;
  daysRemaining?: number;
}

export interface Subscription {
  id?: number;
  email: string;
  stripeSessionId?: string;
  stripeSubscriptionId?: string;
  status: string;
  planName?: string;
  startDate?: string;
  expiresAt?: string;
  createdAt?: string;
  updatedAt?: string;
}
